src/lib/pipeline/ExecuteUnit.cc

```diff
@@ -222,6 +222,15 @@ uint64_t ExecuteUnit::getBranchMispredictedCount() const {
   return branchMispredicts_;
 }
 
+bool ExecuteUnit::isEmpty() {
+  // Execution unit is considered empty if no instructions are present in the
+  // pipeline_ and operationsStalled_ queues
+  if (pipeline_.size() != 0 || operationsStalled_.size() != 0) {
```

Comment:
can we not write `return !(pipeline_.size() != 0 || operationsStalled_.size() != 0)`